src/test/java/org/springframework/data/hazelcast/repository/query/HazelcastSortAccessorTest.java

```diff
@@ -0,0 +1,161 @@
+package org.springframework.data.hazelcast.repository.query;
+
+import com.hazelcast.query.Predicate;
+import org.junit.Test;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.history.RevisionSort;
+import org.springframework.data.keyvalue.core.query.KeyValueQuery;
+import org.springframework.util.ObjectUtils;
+
+import java.io.Serializable;
+import java.util.*;
+
+import static org.junit.Assert.*;
+
+public class HazelcastSortAccessorTest {
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testResolvingNonNativeNullHandlingOrderIsUnsupported() {
+        // given
+        KeyValueQuery<String> keyValueQuery = new KeyValueQuery<String>("^pg~]=P");
+        Sort.Order[] sortOrderArray = new Sort.Order[7];
+        Sort.Order sortOrderA = Sort.Order.asc("^pg~]=P");
+        sortOrderArray[0] = sortOrderA;
+        Sort.Order sortOrderB = sortOrderA.nullsLast();
+        sortOrderArray[1] = sortOrderB;
+        Sort sort = Sort.by(sortOrderArray);
+        keyValueQuery.setSort(sort);
```

Comment:
That section looks overcomplicated for no reason, why not just do the following?

```suggestion
        KeyValueQuery<String> keyValueQuery = new KeyValueQuery<>("^pg~]=P");
        Sort.Order sortOrder1 = Sort.Order.asc("^pg~]=P");
        Sort.Order sortOrder2 = sortOrder1.nullsLast();
        keyValueQuery.setSort(Sort.by(sortOrder1, sortOrder2));
```